src/POData/UriProcessor/UriProcessorNew.php

```diff
@@ -331,6 +330,42 @@ protected function executePost()
                 }
 
                 $payload = $this->getRequest()->getData();
+                if ($payload instanceof ODataURL) {
+                    try {
+                        $this->executeGet();
+                        //dd($this->getRequest()->getSegments()[1]);
+                        $masterModel = $this->getRequest()->getSegments()[0]->getResult();
+                        $masterResourceSet = $this->getRequest()->getSegments()[0]->getTargetResourceSetWrapper();
+                        $masterNavProperty = $this->getRequest()->getLastSegment()->getIdentifier();
+                        $slaveModelUri = new \POData\Common\Url($payload->url);
+                        $host = $this->service->getHost();
+                        $absoluteServiceUri = $host->getAbsoluteServiceUri();
+                        $requestUriSegments = array_slice(
+                        $slaveModelUri->getSegments(),
+                        $absoluteServiceUri->getSegmentCount()
+                    );
+                        $newSegments = \POData\UriProcessor\ResourcePathProcessor\SegmentParser\SegmentParser::parseRequestUriSegments(
```

Comment:
Can you add a using statement to clean this up and fix the PSR-2 break on line length?